registry/remote/example_test.go

```diff
@@ -0,0 +1,542 @@
+func ExampleRepository_Push() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	mediaType, content := ocispec.MediaTypeImageLayer, []byte("Example layer content") // Setup input: 1) media type and 2)[]byte content
+	descriptor := ocispec.Descriptor{                                                  // Assemble a descriptor
+		MediaType: mediaType,                    // Set mediatype
+		Digest:    ocidigest.FromBytes(content), // Calculate digest
+		Size:      int64(len(content)),          // Include content size
+	}
+	err = repo.Push(ctx, descriptor, bytes.NewReader(content)) // Push the blob
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	fmt.Println("Push finished")
+	// Output:
+	// Push finished
+}
+
+// ExampleRepository_Resolve_byTag gives example snippets for resolving a tag to a manifest descriptor.
+func ExampleRepository_Resolve_byTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	descriptor, err := repo.Resolve(ctx, tag) // Resolve the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(descriptor.MediaType)
+	fmt.Println(descriptor.Digest)
+	fmt.Println(descriptor.Size)
+
+	// Output:
+	// application/vnd.oci.image.manifest.v1+json
+	// sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b
+	// 24
+}
+
+// ExampleRepository_Resolve_byDigest gives example snippets for resolving a digest.
+func ExampleRepository_Resolve_byDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+	digest := "sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b"
+	descriptor, err := repo.Resolve(ctx, digest) // Resolve digest to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(descriptor.MediaType)
+	fmt.Println(descriptor.Digest)
+	fmt.Println(descriptor.Size)
+
+	// Output:
+	// application/vnd.oci.image.manifest.v1+json
+	// sha256:00e5ffa7d914b4e6aa3f1a324f37df0625ccc400be333deea5ecaa199f9eff5b
+	// 24
+}
+
+// ExampleRepository_Fetch_byTag gives example snippets for downloading a manifest by tag.
+func ExampleRepository_Fetch_manifestByTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
```

Comment:
Does this actually do any request. My understanding was that this is just parsing and creating a new repo object. Recommend removing the comment since the `.Repositry` initialization might be self explanatory. 
```suggestion
        repo, err := reg.Repository(ctx, exampleRepositoryName)
```


Similarly recommend reducing the comments at the end of the line unless really needed. 
Also would recommend moving comments to above the statements since the rendering is really wide in some cases. 

![image](https://user-images.githubusercontent.com/1821104/162864592-e583ae98-d5c3-4dca-a26a-ad3964e1337a.png)

Reply:
Removed unnecessary comments at the end of the line to make the doc looks better in HTML